languages/rust/oso/tests/test_oso.polar

```diff
@@ -0,0 +1,27 @@
+allow(actor, action, resource) if
+    allowRole(role, action, resource) and
+    actorInRole(actor, role, resource);
+
+allow(_: {sub: sub}, action, resource) if
+    allow(new Actor(name: sub), action, resource);
+
+allow("guest", action, resource) if
+    allow(new Actor(name: "guest"), action, resource);
+
+allow(_: {username: name}, action, resource) if
+    allow(new Actor(name: name), action, resource);
+
+allow(_actor: Actor, "get", _resource: Widget);
+allow(actor: Actor, "create", resource: Company) if
+    resource.role(actor) = "admin";
+
+allow(actor: Actor, "frob", resource: Company) if
+    resource in actor.companies();
+
+# XXX: test_oso::test_allow_models fails with this, but ...
```

Comment:
The top rule, copied over from `test_oso.polar` in the Python tests, fails `test_allow_model` with:

```
Error: TypeError(TypeError { got: Some("oso::host::Class"), expected: "Company" })
thread 'test_allow_model' panicked at 'assertion failed: `(left == right)`
```

But the bottom succeeds. Not sure what I'm doing wrong.

Reply:
This is checking that the input is the _type_ `Company`, as opposed to an instance of a `Company`.

I'm not sure we have a suitable equivalent for this in Rust. It would either have to be by implementing `ToPolar` for `TypeId` so that you could do `TypeId::of::<Company>()` as the input value. Or using the built-in `Company::get_polar_class()`.

Reply:
On reflection, if the latter works then that is probably the best option for now.

Reply:
I removed the comments and commented rule.